projects/igniteui-angular/src/lib/select/select.component.ts

```diff
@@ -568,18 +568,19 @@ export class IgxSelectComponent extends IgxDropDownComponent implements IgxSelec
     }
 
     protected onStatusChanged() {
-        if ((this.ngControl.control.touched || this.ngControl.control.dirty) &&
-            (this.ngControl.control.validator || this.ngControl.control.asyncValidator)) {
-            if (this.inputGroup.isFocused) {
-                this.input.valid = this.ngControl.invalid ? IgxInputState.INVALID : IgxInputState.VALID;
+        this.manageRequiredAsterisk();
+        if ((this.ngControl.control.touched || this.ngControl.control.dirty)) {
```

Comment:
Same as the other ones - try to make this a bit easier to follow by separating the different checks into getters/util functions, like you've done [here](https://github.com/IgniteUI/igniteui-angular/blob/ae93fc19c3c0a425bc4da3f89a6583b5a3846db8/projects/igniteui-angular/src/lib/date-picker/date-picker.component.ts#L855)